### PR TITLE
Implement parts of java.util.logging

### PR DESCRIPTION
**Pull request by cquiroz**
This PR contains an implementation of large part of the java util logging package. This PR was submitted to scala-js as scala-js/scala-js#2410

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # scalajs-java-logging
 
 [![Build Status](https://travis-ci.org/scala-js/scala-js-java-logging.svg?branch=master)](https://travis-ci.org/scala-js/scala-js-java-logging)
-[![Scala.js](https://www.scala-js.org/assets/badges/scalajs-0.6.9.svg)](https://www.scala-js.org/)
+[![Scala.js](https://www.scala-js.org/assets/badges/scalajs-0.6.8.svg)](https://www.scala-js.org/)
```
Comment (nicolasstucki) [inline README.md]: You should not change this.

File: src/main/scala/java/util/logging/Logger.scala [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  }
+
+  def getGlobal(): Logger = globalLogger
+
+  def getLogger(name: String): Logger = {
+    if (name == null)
+      throw new NullPointerException("Logger name cannot be null")
+
+    loggers.getOrElseUpdate(name, {
+      val l = new Logger(name, null)
```
Comment (nicolasstucki) [inline src/main/scala/java/util/logging/Logger.scala]: 4 spaces

Comment (nicolasstucki) [inline src/main/scala/java/util/logging/Logger.scala]: Actually define `l` outside `loggers.getOrElseUpdate`.

``` scala
def newLogger = {
  val logger = new Logger(name, null)
  logger.setLevel(null)
  logger.setUseParentHandlers(true)
  logger.setParent(rootLogger)
  logger
}

loggers.getOrElseUpdate(name, newLogger)
```

Comment (nicolasstucki) [inline src/main/scala/java/util/logging/Logger.scala]: You could also put `newLogger` as a private method of the class add a `name` parameter and use it to implement `getAnonymousLogger`.

File: src/main/scala/java/util/logging/Logger.scala [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+  def setFilter(filter: Filter): Unit = this.filter = filter
+
+  def getFilter(): Filter = filter
+
+  def log(record: LogRecord): Unit = {
+    if (isLoggable(record.getLevel)) {
+      if (useParentsHandlers) {
+        val parent = getParent()
+        if (parent != null) parent.log(record)
```
Comment (nicolasstucki) [inline src/main/scala/java/util/logging/Logger.scala]: `parent.log(record)` on the next line. We avoid putting statements on the same line of the `if`, we only do it for expressions.

**Comment by nicolasstucki** [comment]
That is all. Otherwise LGTM.

File: README.md (hunk 1 above)
Comment (cquiroz) [inline README.md]: The svg for 6.9 is missing. I guess the solution is to add that badge to scala-js

**Comment by cquiroz** [comment]
Is it ok to add my name to the list of developers?

File: README.md (hunk 1 above)
Comment (sjrd) [inline README.md]: @cquiroz You are actually right to change it to 0.6.8. But ideally that would be in a separate PR, as it irrelevant to this PR.

**Comment by sjrd** [comment]
> Is it ok to add my name to the list of developers?

Good question. According to https://maven.apache.org/pom.html#Developers, I don't think you should. They say:

> Developers are presumably members of the project's core development. Note that, although an organization may have many developers (programmers) as members, it is not good form to list them all as developers, but only those who are immediately responsible for the code. A good rule of thumb is, if the person should not be contacted about the project, they need not be listed here.

From that paragraph, I interpret the term "developer" more as "maintainer", i.e., people who are in charge of the repository.

If we had a list of "contributors", that would be a good spot to add your name. As it happens, we have a perfectly accurate and up to date contributor list at https://github.com/scala-js/scala-js-java-logging/graphs/contributors, where you'll clearly appear as the main contributor as soon as this PR gets merged :-)

**Comment by cquiroz** [comment]
Updated with comments from @nicolasstucki

File: testSuite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/logging/LogRecordTest.scala [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    assertEquals(1L, record.getMillis)
+
+    assertNull(record.getThrown)
+    record.setThrown(new RuntimeException("testmsg"))
+    assertEquals("testmsg", record.getThrown.getMessage)
+    record.setThrown(null)
+    assertNull(record.getThrown)
+  }
+
+
```
Comment (nicolasstucki) [inline testSuite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/logging/LogRecordTest.scala]: 2 lines, remove one

File: src/main/scala/java/util/logging/Formatter.scala [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          // Beginning of param
+          inParam = true
+          paramInFlight = new StringBuilder()
+        } else if (inParam && currentChar != '}') {
+          // accumulate the param
+          paramInFlight += currentChar
+        } else if (currentChar == '}') {
+          // end of param, replace placeholder by value if possible
+          inParam = false
+          val (failed, replacement) =
```
Comment (nicolasstucki) [inline src/main/scala/java/util/logging/Formatter.scala]: Add `{ }` arround the multiline `try`

```
val (failed, replacement) = {
  try {
    ...
  } catch {
    ...
  }
}
```

**Comment by nicolasstucki** [comment]
That is all.

**Comment by cquiroz** [comment]
I'm squashing my updates right away, but perhaps it is easier to review if I only commit the fixes and the squash at the end. Any preference?

**Comment by nicolasstucki** [comment]
It is fine if you squash.

**Comment by cquiroz** [comment]
Updated with the last comments, thanks!

**Comment by nicolasstucki** [comment]
LGTM
